src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelCatalogQueries.scala

```diff
@@ -193,8 +208,22 @@ object ReadModelCatalogQueries extends ReadModelQuery {
 
     val modeFilter = mode.map(_.toString).map(Filters.eq("data.mode", _))
 
+    val visibilityRestrictionsFilter =
+      if (visibilityRestrictions)
+        Some(
+          Filters.nor(
+            Filters.and(Filters.ne("data.producerId", requesterId.toString), Filters.size("data.descriptors", 0)),
+            Filters.and(
+              Filters.ne("data.producerId", requesterId.toString),
+              Filters.size("data.descriptors", 1),
+              Filters.eq("data.descriptors.state", Draft.toString())
+            )
+          )
+        )
+      else Some(Filters.empty())
```

Comment:
same result, but without adding an empty condition in the query
```suggestion
      else None
```